src/server/rdb_save.cc

```diff
@@ -263,7 +266,9 @@ error_code RdbSerializer::SelectDb(uint32_t dbid) {
 
 // Called by snapshot
 io::Result<uint8_t> RdbSerializer::SaveEntry(const PrimeKey& pk, const PrimeValue& pv,
-                                             uint64_t expire_ms) {
+                                             uint64_t expire_ms, DbIndex dbid) {
+  SelectDb(dbid);
+  last_entry_db_index_ = dbid;
```

Comment:
is SelectDb used anywhere else?  
anyway, I think this line should be moved to  SelectDb -line 261

Reply:
true this is left overs from my implementation before some fix, now its not relevant. will move inside